### PR TITLE
React error boundaries

### LINKED ISSUE #5720: React error boundaries
We won't catch every runtime client-side React bug in advance. 
When it happens, we should ideally be sending the exception to something like Sentry, but we should also have a "pretty" error page like we have in Django for 500.html errors. 

This is what it looks like if an error happens (in this example, in `article.jsx`)

Screen Shot 2019-08-28 at 2 08 43 PM

If it's not weird enough already, it'll be even worse once the footer is a React component.

### LINKED ISSUE #5913: Unit test error boundaries
https://github.com/mozilla/kuma/issues/5720 is about adding an error boundary. We should have some unit tests of these.

### PR DESCRIPTION
**Pull request by peterbe**
Fixes #5720 
Fixes #5913

When the whole app fails:
Screen Shot 2019-09-26 at 12 04 31 PM

When the `Content` component fails:
Screen Shot 2019-09-26 at 12 04 45 PM

### CODE REVIEW

**Comment by peterbe** [comment]
Note, I didn't bother with Flow. Perhaps this can be a follow-up issue. 
Also, no unit tests. Also a follow-up issue.

**Comment by atopal** [comment]
Can we also get a GA error for these cases? It's much easier to work with that for me.

**Comment by peterbe** [comment]
> Can we also get a GA error for these cases? It's much easier to work with that for me.

Just like we have a separate issue just for Sentry sending, I can make a separate issue for doing a GA sending thing.

**Comment by peterbe** [comment]
I think I cut corners on this one and I don't like it. I shouldn't give up on Flow and jest testing. We have a deadline but I don't know if it's fair to block that on better error reporting. 

I'm going to take this back and try to add...

* Flow
* jest tests
* GA tracker

**Comment by peterbe** [comment]
@Gregoor We had the MDN Dev sprint planning meeting and chatted about this. Time is of the essence and the team had me convinced. 

I'm going to write follow-up issues that worries about using GA tracker and one that worries about Flow and one that  worries about a jest unit test of the component. 

The truth is, without this PR landed as soon as possible, we won't have any Sentry on client-side errors. And we generally feel like we need tthis to launch the React site with confidence.

File: kuma/javascript/src/index.jsx [hunk 1]
```diff
@@ -33,88 +34,104 @@ if (container) {
...
-                        initialURL={data.url}
-                        initialData={data.documentData}
-                    />
-                </React.StrictMode>
+                <SinglePageApp
+                    initialURL={data.url}
+                    initialData={data.documentData}
+                />
             );
+            // app = (
```
Comment (Gregoor) [inline kuma/javascript/src/index.jsx]: I like what you've done with moving StrictMode-wrapping downstairs, but I'd vote for also deleting the commented out code.

File: kuma/javascript/src/error-boundaries.jsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    componentDidCatch(error, errorInfo) {
+        document.title = 'Application rendering Error';
+        this.logError(error, errorInfo);
+    }
+
+    render() {
+        if (this.state.error) {
+            return (
+                <ErrorWrapper title={gettext('Application rendering error')}>
+                    <p>
```
Comment (Gregoor) [inline kuma/javascript/src/error-boundaries.jsx]: nit: Is there a reason for why we're supplying children to this component when they're the same in both instances?

**Comment by peterbe** [comment]
> Can we also get a GA error for these cases? It's much easier to work with that for me.

https://github.com/mozilla/kuma/issues/5929

File: kuma/javascript/src/error-boundaries.jsx (hunk 2 above)
Comment (peterbe) [reply kuma/javascript/src/error-boundaries.jsx]: No reason. I think I had bigger ideas and intentions but I've run out if ideas :)
Also, using the same English sentence for application errors and content errors makes for *fewer* things that need to be translated. I'm going to move this to the `<ErrorWrapper/>`

By the way, what do you think about the name `ErrorWrapper`?

File: kuma/javascript/src/index.jsx (hunk 1 above)
Comment (peterbe) [reply kuma/javascript/src/index.jsx]: This is one of the many reasons why code reviews are good.

**Pull request merged**